### PR TITLE
Rematch E05620

### PR DESCRIPTION
**Pull request by Ryan-Myers**
Just submitting a new one as merging the last one wasn't worth the effort for something this small.

### CODE REVIEW

**Review by sonicdcer — APPROVED** [review]
Looks good :)

File: src/overlays/ovl_i1/E05620.c [hunk 1]
```diff
@@ -1,7 +1,59 @@
...
+                    func_800BA808(gMsg_ID_20330, RCID_ROB64);
+                    return;
+                }
+            }
+            return;
+        case 1:
+            item->obj.pos.x += ((gPlayer[item->unk_4E].pos.x - item->obj.pos.x) * 0.05f);
+            item->obj.pos.y += ((gPlayer[item->unk_4E].pos.y - item->obj.pos.y) * 0.05f);
+            if (gPlayer->unk_238 != 0) {
+                item->obj.pos.z += (((gPlayer[item->unk_4E].unk_138 - 300.0f) - item->obj.pos.z) * 0.05f);
```
Comment (petrie911) [inline src/overlays/ovl_i1/E05620.c:48]:
```suggestion
                item->obj.pos.z += (gPlayer[item->unk_4E].unk_138 - 300.0f)- item->obj.pos.z) * 0.05f;
```

File: src/overlays/ovl_i1/E05620.c [hunk 2]
```diff
@@ -1,7 +1,59 @@
...
+            if (item->unk_4C != 0) {
+                item->unk_46 = 1;
+                item->timer_48 = 50;
+                item->info.unk_10 = 10000.0f;
+                PRINTF("♪:リング通過音\n"); // Ring passing sound
+                Audio_PlaySfx(0x49008025, &D_800C5D28, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                D_80177E80++;
+                if ((item->obj.rot.x != 0.0f) && (D_80177E80 >= 100)) {
+                    func_800BA808(gMsg_ID_20330, RCID_ROB64);
+                    return;
```
Comment (petrie911) [inline src/overlays/ovl_i1/E05620.c:40]:
```suggestion
```
this shouldn't be needed

File: src/overlays/ovl_i1/E05620.c [hunk 3]
```diff
@@ -1,7 +1,59 @@
...
+                }
+            }
+            return;
```
Comment (petrie911) [inline src/overlays/ovl_i1/E05620.c:43]: can this be break?

Comment (Ryan-Myers) [reply src/overlays/ovl_i1/E05620.c]: Yeah, woops. I forgot to do that after M2C made it a return. Fixed now.

**Pull request merged**